Add noqa to exsiting files and enable all bugbear rules

justinchuby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #100336
* __->__ #100342
* #100341
* #100338
* #100339

Add noqa to exsiting files and enable all bugbear rules so that we can enable all bugbear rules on new code.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @anijain2305 @soumith @desertfire

justinchuby committed:
Add noqa to exsiting files and enable all bugbear rules

[ghstack-poisoned]

justinchuby committed:
Update on "Add noqa to exsiting files and enable all bugbear rules"


Add noqa to exsiting files and enable all bugbear rules so that we can enable all bugbear rules on new code.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Skylion007 commented on pyproject.toml:
```diff
@@ -26,13 +26,7 @@ target-version = "py38"
 # NOTE: Synchoronize the ignores with .flake8
 ignore = [
-    # these ignores are from flake8-bugbear; please fix!
-    "B008", "B017",
```
Everything B01 and above 7 should probably be fixed instead of ignored. I just added those ignores so that they could be fixed manually eventually when I updated bugbear.

justinchuby replied:
These are the rules where autofix is not available. I think it's more realistic to (1) find and remove the noqa when more auto fix is implemented in ruff and do a batch fix and/or (2) Allow code owners to fix cases by case when they see the noqa (as a more obvious reminder in code).

Skylion007 replied:
> These are the rules where autofix is not available.

Yet, ruff is rapidly adding many autofixes and if you file an issue they will often add it for simple fixes (like B007). 

If we do this, we need to differentiate these noqa from actual intentional noqas with some additional text in the comment (pretty sure you can either prepend or append text to the noqa and it will still work, but I forgot which). A useful feature for having a type ignore and noqa on the same line.

If I recall, some of these error codes have 20 or less violations and should be fixed manually. Some of the more involved ones will take singificantly more work for sure.

Skylion007 replied:
https://github.com/charliermarsh/ruff/issues/4163 B027 should be trivial to add.

justinchuby committed:
Update on "Add noqa to exsiting files and enable all bugbear rules"


Add noqa to exsiting files and enable all bugbear rules so that we can enable all bugbear rules on new code.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

Skylion007 commented on torch/ao/quantization/observer.py:
```diff
@@ -925,7 +925,7 @@ def __init__(
         quant_max=None,
         factory_kwargs=None,
-        eps=torch.finfo(torch.float32).eps,
+        eps=torch.finfo(torch.float32).eps,  # noqa: B008
```
A lot of these B008 are false positives and this rule should definitely remain disabled.

justinchuby committed:
Update on "Add noqa to exsiting files and enable all bugbear rules"


Add noqa to exsiting files and enable all bugbear rules so that we can enable all bugbear rules on new code.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]
